yorkie/src/main/kotlin/dev/yorkie/document/crdt/RgaTreeSplit.kt

```diff
@@ -0,0 +1,577 @@
+            }
+            prev = current
+            node = node.next
+        }
+        return clone
+    }
+
+    fun toJson(): String {
+        return buildString {
+            this@RgaTreeSplit.forEach { node ->
+                if (!node.isRemoved) append(node.value)
+            }
+        }
+    }
+
+    companion object {
+        private val InitialNodeID = RgaTreeSplitNodeID(InitialTimeTicket, 0)
+        private const val INITIAL_NODE_VALUE = ""
+    }
+}
+
+internal data class RgaTreeSplitNode<T : CharSequence>(
+    val id: RgaTreeSplitNodeID,
+    private var _value: T,
+    private var _removedAt: TimeTicket? = null,
+) {
+    var prev: RgaTreeSplitNode<T>? = null
+        private set
+    var next: RgaTreeSplitNode<T>? = null
+        private set
+    var insertionPrev: RgaTreeSplitNode<T>? = null
+        private set
+    var insertionNext: RgaTreeSplitNode<T>? = null
+        private set
+
+    val createdAt: TimeTicket
+        get() = id.createdAt
+
+    val length: Int
+        get() = _removedAt?.let { contentLength } ?: 0
+
+    val contentLength: Int
+        get() = _value.length
+
+    val hasNext: Boolean
+        get() = next != null
+
+    val hasInsertionPrev: Boolean
+        get() = insertionPrev != null
+
+    val isRemoved: Boolean
+        get() = _removedAt != null
+
+    val removedAt: TimeTicket?
+        get() = _removedAt
+
+    @VisibleForTesting
+    val value
+        get() = _value
+
+    fun setPrev(node: RgaTreeSplitNode<T>?) {
+        prev = node
+        node?.next = this
+    }
+
+    fun setNext(node: RgaTreeSplitNode<T>?) {
+        next = node
+        node?.prev = this
+    }
+
+    fun setInsertionPrev(node: RgaTreeSplitNode<T>?) {
+        insertionPrev = node
+        node?.insertionNext = this
+    }
+
+    fun setInsertionNext(node: RgaTreeSplitNode<T>?) {
+        insertionNext = node
+        node?.insertionPrev = this
+    }
+
+    /**
+     * Creates a new split node of the given [offset].
+     */
+    fun split(offset: Int): RgaTreeSplitNode<T> {
+        return RgaTreeSplitNode(id.split(offset), splitValue(offset), _removedAt)
+    }
+
+    @Suppress("UNCHECKED_CAST")
+    private fun splitValue(offset: Int): T {
+        val valueBefore = _value
+        _value = valueBefore.subSequence(0, offset) as T
+        return valueBefore.subSequence(offset, valueBefore.length) as T
+    }
+
+    // NOTE(7hong13): original comment from JS-SDK:
+    // `canDelete` checks if node is able to delete.
+    /**
+     * Checks if this [RgaTreeSplitNode] can be deleted or not.
+     */
+    fun canDelete(executedAt: TimeTicket, latestCreatedAt: TimeTicket): Boolean {
+        return createdAt <= latestCreatedAt && (isRemoved || _removedAt < executedAt)
+    }
+
+    // NOTE(7hong13): original comment from JS-SDK:
+    // `remove` removes node of given edited time.
+    /**
+     * Removes this [RgaTreeSplitNode] at the given [executedAt].
```

Comment:
@hackerwins I left some notes on the comments I fixed.
(refer to: [link](https://github.com/yorkie-team/yorkie-js-sdk/blob/893b2c9a870129068d152e092b2908b53fef8e72/src/document/crdt/rga_tree_split.ts#L406))

I will remove all the notes in `RgaTreeSplit` in next PR.